src/hardware/mixer.cpp

```diff
@@ -940,116 +1007,165 @@ static constexpr int calc_tickadd(const int freq)
+			        mixer.work[pos][ch]);
+		}
+		pos = (pos + 1) & MIXER_BUFMASK;
+	}
 
 	if (CaptureState & (CAPTURE_WAVE | CAPTURE_VIDEO)) {
-		int16_t convert[1024][2];
-		const auto added = check_cast<work_index_t>(std::min(needed - mixer.done, 1024));
-		auto readpos = check_cast<work_index_t>((mixer.pos + mixer.done) & MIXER_BUFMASK);
+		int16_t out[capture_buf_len][2];
+
+		auto pos = check_cast<work_index_t>(
+		        (mixer.pos + mixer.frames_done) & MIXER_BUFMASK);
+
 		for (work_index_t i = 0; i < added; i++) {
-			const auto sample_1 = mixer.work[readpos][0];
-			const auto sample_2 = mixer.work[readpos][1];
-			const auto s1 = static_cast<uint16_t>(MIXER_CLIP(static_cast<int>(sample_1)));
-			const auto s2 = static_cast<uint16_t>(MIXER_CLIP(static_cast<int>(sample_2)));
-			convert[i][0] = static_cast<int16_t>(host_to_le16(s1));
-			convert[i][1] = static_cast<int16_t>(host_to_le16(s2));
-			readpos = (readpos + 1) & MIXER_BUFMASK;
+			const auto left = static_cast<uint16_t>(
+			        MIXER_CLIP(static_cast<int>(mixer.work[pos][0])));
+
+			const auto right = static_cast<uint16_t>(
+			        MIXER_CLIP(static_cast<int>(mixer.work[pos][1])));
+
+			out[i][0] = static_cast<int16_t>(host_to_le16(left));
+			out[i][1] = static_cast<int16_t>(host_to_le16(right));
+
+			pos = (pos + 1) & MIXER_BUFMASK;
 		}
-		CAPTURE_AddWave(mixer.sample_rate, added, reinterpret_cast<int16_t*>(convert));
+		CAPTURE_AddWave(mixer.sample_rate,
+		                added,
+		                reinterpret_cast<int16_t *>(out));
 	}
-	//Reset the the tick_add for constant speed
-	if( Mixer_irq_important() )
+	// Reset the the tick_add for constant speed
+	if (Mixer_irq_important())
 		mixer.tick_add = calc_tickadd(mixer.sample_rate);
-	mixer.done = needed;
+
+	mixer.frames_done = frames_requested;
 }
 
 static void MIXER_Mix()
 {
 	MIXER_LockAudioDevice();
-	MIXER_MixData(mixer.needed);
+	MIXER_MixData(mixer.frames_needed);
 	mixer.tick_counter += mixer.tick_add;
-	mixer.needed+=(mixer.tick_counter >> TICK_SHIFT);
+	mixer.frames_needed += mixer.tick_counter >> TICK_SHIFT;
 	mixer.tick_counter &= TICK_MASK;
 	MIXER_UnlockAudioDevice();
 }
 
 static void MIXER_ReduceChannelsDoneCounts(const int at_most)
 {
 	for (auto &it : mixer.channels)
-		it.second->done -= std::min(it.second->done.load(), at_most);
+		it.second->frames_done -= std::min(it.second->frames_done.load(),
+		                                   at_most);
 }
 
 static void MIXER_Mix_NoSound()
 {
 	MIXER_LockAudioDevice();
-	MIXER_MixData(mixer.needed);
+	MIXER_MixData(mixer.frames_needed);
+
 	/* Clear piece we've just generated */
-	for (auto i = 0; i < mixer.needed; ++i) {
-		mixer.work[mixer.pos][0]=0;
-		mixer.work[mixer.pos][1]=0;
-		mixer.pos=(mixer.pos+1)&MIXER_BUFMASK;
+	for (auto i = 0; i < mixer.frames_needed; ++i) {
+		mixer.work[mixer.pos][0] = 0;
+		mixer.work[mixer.pos][1] = 0;
+		mixer.pos                = (mixer.pos + 1) & MIXER_BUFMASK;
```

Comment:
> I do like the new `= value` alignment we're using in Clang.. and what it's shown me is that visually, I need to add new line to differentiate categories of types, otherwise everything is block-aligned, which gives the impression the values fall into the same "grouping"

Indeed, I like it for the same reason — it forces you to use meaningful groupings.